domain/src/main/kotlin/com/krzykrucz/elesson/currentlesson/domain/attendance/AttendanceWorkflowImplementation.kt

```diff
@@ -0,0 +1,76 @@
+package com.krzykrucz.elesson.currentlesson.domain.attendance
+
+import arrow.core.Either
+import arrow.core.left
+import arrow.core.right
+import com.krzykrucz.elesson.currentlesson.domain.startlesson.ClassRegistry
+import com.krzykrucz.elesson.currentlesson.domain.startlesson.LessonIdentifier
+import java.time.LocalDateTime
+
+fun noteAbsence(
+        isLessonStarted: IsLessonStarted,
+        isInRegistry: IsInRegistry,
+        areAllStudentsChecked: AreAllStudentsChecked
+): NoteAbsence = { lessonId, uncheckedStudent, attendanceList, classRegistry ->
+    noteStudent(isLessonStarted, lessonId, isInRegistry, uncheckedStudent, classRegistry, attendanceList, areAllStudentsChecked, addAbsentStudent())
+}
+
+fun notePresence(
+        isLessonStarted: IsLessonStarted,
+        isInRegistry: IsInRegistry,
+        areAllStudentsChecked: AreAllStudentsChecked
+): NotePresence = { lessonId, uncheckedStudent, attendanceList, classRegistry ->
+    noteStudent(isLessonStarted, lessonId, isInRegistry, uncheckedStudent, classRegistry, attendanceList, areAllStudentsChecked, addPresentStudent())
+}
+
+fun noteLate(
+        isNotTooLate: IsNotTooLate
+): NoteLate = { lessonId, absentStudent, completedAttendance ->
+    if (isNotTooLate(lessonId.lessonStartTime, LocalDateTime.now())) {
```

Comment:
let's assume it's too late to note student present 15 minutes into the lesson scheduled time (regardless when it really started), so lesson hour number should be taken into account, not start time

in such case it's unnecessary to add start time to LessonIdentifier

Reply:
could you also implement this isNotTooLate function?

Reply:
I got confused when it comes to lesson hour number, maybe ScheduledLessonStart would be more descriptive?

Yeah, no problem.